src/jit/morph.cpp

```diff
@@ -83,6 +83,18 @@ GenTreePtr          Compiler::fgMorphIntoHelperCall(GenTreePtr      tree,
     tree->gtCall.gtEntryPoint.addr = nullptr;
 #endif
 
+#if defined(_TARGET_X86_) && !defined(LEGACY_BACKEND)
+    if (varTypeIsLong(tree))
+    {
+        GenTreeCall* callNode = tree->AsCall();
+        ReturnTypeDesc* retTypeDesc = callNode->GetReturnTypeDesc();
+        retTypeDesc->Reset();
+        retTypeDesc->Initialize(this, callNode->gtRetClsHnd);
+        
+        NYI("Helper with TYP_LONG return type");
+    }
+#endif
+
```

Comment:
I'm curious why this is needed; is there something different about the helper call case, as compared with the generic call case that you are now handling? Is this because they will not have been forced to a var=call by the code that handles multi-reg structs? If so, then I'm wondering why the lowering code needs to handle the case where it is not already var=call. I suspect there's something I'm missing ...